GooglePlus profile image support

anotherdirtbag opened the pull request:
adding support for getting a googleplus thumbnail image from their email. This is very similar to the existing gravitar support but can fail if a non-google account or invalid email is used. I had it fall back to gravitar and picture in that case.

anotherdirtbag committed:
adding support for googleplus thumbnail image

if a google email address is added to googleplus on device init, then config_picture will be set to the thumbnail of their current googleplus profile image

anotherdirtbag committed:
adding unit tests for googleplus images

homeassistant commented:
Hi @anotherdirtbag, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#10281).

Once you do that we will be able to review and accept this pull request.

Thanks!

Danielhiversen commented:
Please fill out the template for the pull request.
There also som style issues with your code.

anotherdirtbag committed:
updated googleplus handling

gravatar is now checked before googleplus
google email now uses url parsing to encode non-ascii characters

homeassistant commented:
Hi @anotherdirtbag, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#10281).

Once you do that we will be able to review and accept this pull request.

Thanks!

anotherdirtbag committed:
updated googleplus unit tests

gravatar is now preferred over googleplus
fixes some whitespace

homeassistant commented:
Hi @anotherdirtbag, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#10281).

Once you do that we will be able to review and accept this pull request.

Thanks!

anotherdirtbag committed:
fixed whitespace

homeassistant commented:
Hi @anotherdirtbag, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant#10281).

Once you do that we will be able to review and accept this pull request.

Thanks!

anotherdirtbag commented:
crap. so i committed the wrong file into my branch and i don't know enough git to undo that. I'm going to start over from a new fork later.

Danielhiversen commented:
Ok, then I close this pr.
Remember to fill in the pull request template next time.
